drivers/pwm/pwm_mcux_sctimer.c

```diff
@@ -72,42 +118,82 @@ static int mcux_sctimer_pwm_set_cycles(const struct device *dev,
 			base->OUTPUT |= (1UL << channel);
 		}
 
+		/* Set duty cycle to 0 for this channel */
+		SCTIMER_UpdatePwmDutycycle(base, channel, duty_cycle,
+					   data->event_number[channel]);
+		if (data->configured_chan > 1) {
+			/* Make sure the global counter gets restarted */
+			SCTIMER_StartTimer(base, kSCTIMER_Counter_U);
+		} else {
+			/* Only one channel setup, we can safely stop counter.
+			 * we must call this again because the duty cycle
+			 * update function will have started it
+			 */
+			SCTIMER_StopTimer(base, kSCTIMER_Counter_U);
+		}
```

Comment:
If there is not duty cycle, then signal can be driven high or low directly. 
Why is this additional code required?

Reply:
Sorry- I'm a bit unclear here. Are you asking why we start/stop the SCTimer at the end of this function? Or why we update the duty cycle after setting the output?

Reply:
I was asking about `SCTIMER_UpdatePwmDutycycle`. Since this is done on a per channel basis, isn't this happening for a particular channel which has already been configured earlier by setting the bits in the `OUTPUT` register. 
I see the need to restart the timer, however the `else` case can be deleted as the timer has already been stopped earlier.

Reply:
> I was asking about SCTIMER_UpdatePwmDutycycle. Since this is done on a per channel basis, isn't this happening for a particular channel which has already been configured earlier by setting the bits in the OUTPUT register.

Yes, it is. This is only required in the case where multiple channels are active, since otherwise the channel would keep on sending a PWM signal once the timer was restarted.

I think I will go ahead and update this code to only run when a single channel is configured, based on the comment I made above. It should make the code flow here simpler

Reply:
Setting the bits in the `OUTPUT` register should force the corresponding output `HIGH` or `LOW`. 
Do you see the count registers running affect the output?

Reply:
Hi, I do see the event registers affect the output once the counter is restarted. I believe that the `OUTPUT` registers are only used for forcing the output when the timer is stopped. I tested with this code:

```
/* Set the output to inactive State */
if (data->channel[channel].level == kSCTIMER_HighTrue) {
        base->OUTPUT &= ~(1UL << channel);
} else {
        base->OUTPUT |= (1UL << channel);
}

// TEST- restart the SCTIMER, to see if the OUT bit toggles


while (1) {
        if (data->channel[channel].level == kSCTIMER_HighTrue) {
                if (base->OUTPUT & (1U << channel)) {
                        /* output changed- exit */
                        break;
                }
        } else {
                if ((base->OUTPUT & (1U << channel)) == 0) {
                        /* output changed- exit */
                        break;
                }
        }
}
printk("SCTIMER output changed\n");
printk("SCTIMER output bits: 0x%x\n", base->OUTPUT);
```

And I do see the output changing when the duty cycle is set to 0:

```
[PWM]: 0, [period]: 2000000, [pulse]: 0
SCTIMER output bits: 0x0
SCTIMER output changed
SCTIMER output bits: 0x1
```